stm32 nucleo-g0b1re fixes

raul-klg opened the pull request:
Hello. I'd like to help with some little changes/fixes to the recently added support to stm32g0b1 and nucleo-g0b1re, specifically to the related dts files.

raul-klg committed:
board: arm: Fix nucleo_g0b1re dts warnings.

* Removed leading 0 from slot0_partition starting offset
* Fixed slot1_partition starting offset

This fixes the following warnings:

nucleo_g0b1re.dts.pre.tmp:1749.36-1752.5: Warning
 (unit_address_format):
 /soc/flash-controller@40022000/flash@8000000/partitions/
 partition@0C000:unit name should not have leading 0s
warning: unit address and first address in 'reg' (0x3e000) don't match
 for /soc/flash-controller@40022000/flash@8000000/partitions/
 partition@31000

Signed-off-by: Raúl Sánchez Siles <rsanchezs@k-lagan.com>

raul-klg committed:
board: arm: Fix nucleo_g0b1re arduino connector dts

According to ST documentation for nucleo-g0b1re board:
https://my.st.com/resource/en/schematic_pack/mb1360-g0b1re-c02_schematic.pdf
the D0 and D1 lines in the Arduino UNO connector are mapped to the PC5
and PC4 pins.

Signed-off-by: Raúl Sánchez Siles <rsanchezs@k-lagan.com>

raul-klg committed:
board: arm: Fix nucleo_g0b1re dts compatible

It should be st,stm32g0b1re-nucleo

Signed-off-by: Raúl Sánchez Siles <rsanchezs@k-lagan.com>

erwango reviewed (changes requested):
Thanks for these fixes!
Aside from the compliance checks, looks good.
Please fix the reported long lines in commit messages (which otherwise are well documented, thanks for that also).

str4t0m commented on boards/arm/nucleo_g0b1re/arduino_r3_connector.dtsi:
```diff
@@ -16,8 +16,8 @@
 			   <5 0 &gpiob 12 0>,	/* A5 */
-			   <6 0 &gpioa 3 0>,	/* D0 */
-			   <7 0 &gpioa 2 0>,	/* D1 */
+			   <6 0 &gpioc 5 0>,	/* D0 */
+			   <7 0 &gpioc 4 0>,	/* D1 */
```
The nucleo_g071rb, uses the same board, could you eventually fix it there as well?

str4t0m reviewed (changes requested):
One nit.

jenmwms commented:
> Please fix the reported long lines in commit messages

To answer your question from dev mail list, there are a few ways to satisfy these compliance checks. Here, IMO you can just cut the long line onto another line to satisfy the limit. Line break between the warnings can help readability then. Or, you can file a quick bug and link the number in your commit with a Fixes #xxxxx.  In some cases, when not simple to shorten, it may be permissible based on feedback from approvers.

raul-klg committed:
board: arm: Fix nucleo_g071rb arduino connector dts

According to ST documentation for nucleo-g071rb board:
https://www.st.com/resource/en/schematic_pack/mb1360-g071rb-c01_schematic.pdf
the D0 and D1 lines in the Arduino UNO connector are mapped to the PC5
and PC4 pins.

Signed-off-by: Raúl Sánchez Siles <rsanchezs@k-lagan.com>
